src/frontend/app/features/cloud-foundry/edit-space/edit-space-step/edit-space-step.component.ts

```diff
@@ -0,0 +1,184 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { MatSnackBar } from '@angular/material';
+import { ActivatedRoute } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { Observable } from 'rxjs/Observable';
+import { filter, map, tap, take } from 'rxjs/operators';
+import { Subscription } from 'rxjs/Subscription';
+
+import { EntityServiceFactory } from '../../../../core/entity-service-factory.service';
+import { CfOrgSpaceDataService } from '../../../../shared/data-services/cf-org-space-service.service';
+import { CfUserService } from '../../../../shared/data-services/cf-user.service';
+import { PaginationMonitorFactory } from '../../../../shared/monitors/pagination-monitor.factory';
+import { SpaceSchema, spaceSchemaKey } from '../../../../store/actions/action-types';
+import { GetAllSpacesInOrg } from '../../../../store/actions/organisation.actions';
+import { getPaginationKey } from '../../../../store/actions/pagination.actions';
+import { AppState } from '../../../../store/app-state';
+import { getPaginationObservables } from '../../../../store/reducers/pagination-reducer/pagination-reducer.helper';
+import { APIResource } from '../../../../store/types/api.types';
+import { CloudFoundryEndpointService } from '../../services/cloud-foundry-endpoint.service';
+import { CloudFoundrySpaceService } from '../../services/cloud-foundry-space.service';
+import { UpdateSpace } from '../../../../store/actions/space.actions';
+import { selectRequestInfo } from '../../../../store/selectors/api.selectors';
+import { RouterNav } from '../../../../store/actions/router.actions';
+
+const cfSpaceServiceFactory = (
+  store: Store<AppState>,
+  activatedRoute: ActivatedRoute,
+  entityServiceFactory: EntityServiceFactory,
+  cfOrgSpaceDataService: CfOrgSpaceDataService,
+  cfUserService: CfUserService,
+  paginationMonitorFactory: PaginationMonitorFactory,
+  cfEndpointService: CloudFoundryEndpointService
+) => {
+  const { orgId, spaceId } = activatedRoute.snapshot.params;
+  const { cfGuid } = cfEndpointService;
+  return new CloudFoundrySpaceService(
```

Comment:
If you're fetching org/space id in the component you might as well create this service there too, avoids a lot of lines and c+p between components (could create service here and in parent component the same way)